server/tso/allocator_manager.go

```diff
@@ -36,23 +36,18 @@ import (
 )
 
 const (
+	checkAllocatorStep          = 1 * time.Second
 	dcLocationConfigEtcdPrefix  = "dc-location"
-	leaderTickInterval          = 50 * time.Millisecond
 	defaultAllocatorLeaderLease = 3
+	leaderTickInterval          = 50 * time.Millisecond
 )
 
 // AllocatorGroupFilter is used to select AllocatorGroup.
 type AllocatorGroupFilter func(ag *allocatorGroup) bool
 
 type allocatorGroup struct {
 	dcLocation string
-	// allocator's parent ctx and cancel function, which is to
-	// control the allocator's behavior in AllocatorDaemon and
-	// pass the cancel signal to its parent as soon as possible
-	// since it is critical to let parent goroutine know whether
-	// the allocator is still able to work well.
-	parentCtx    context.Context
-	parentCancel context.CancelFunc
+	serverCtx  context.Context
```

Comment:
It seems `allocatorGroup` and `allocatorManager` holding the same serverCtx passed from the `Server`, so I think it is not necessary for `allocatorGroup` to hold `serverCtx`. And `allocatorUpdater` should pass the `serverCtx` to the `allocatorGroup` in that function.